File: 백준/Silver/10815. 숫자 카드/숫자 카드.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> v;
int binary_search(int start, int end, int target)
{
	while (start <= end)
	{
		int middle = (start + end) / 2;

		if (target < v[middle])
			end = middle - 1;
		else if (target > v[middle])
			start = middle + 1;
		else return middle;
	}

	return -1;
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	int n;
	cin >> n;

	
	for (int i = 0; i < n; i++)
	{
		int input;
		cin >> input;
		v.push_back(input);
	}
	sort(v.begin(), v.end());

	int m;
	cin >> m;
	vector<int> result;
	for (int i = 0; i < m; i++)
	{
		int input;
		cin >> input;

		if (binary_search(0, v.size() - 1, input) == -1)
			result.push_back(0);
		else
			result.push_back(1);
	}

	for (int i = 0; i < result.size(); i++)
	{
		cout << result[i] << " ";
	}
}

File: 백준/Silver/10815. 숫자 카드/숫자 카드.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on('line', (line) => {
    // 입력 받는 공간
    input.push(line.trim());
}).on('close', () => {
    // 문제를 풀고 출력하는 공간
    let n = input.shift().split(' ').map(Number);
    let num1 = input.shift().split(' ').map(Number);
   
    let m = input.shift().split(' ').map(Number);
    let num2 = input.shift().split(' ').map(Number);
    
    num1.sort((a, b) => a - b);
    const binatySearch = (v, start, end, target) => {
        while(start <= end) {
            let middle = Math.floor((start + end) / 2);
            if(v[middle] < target) 
                start = middle + 1;
            else if(v[middle] > target)
                end = middle - 1;
            else
                return 1;
        }
        return 0;
    }

    let result = [];
    for(let i = 0; i < num2.length; i++) {
        result.push(binatySearch(num1, 0, num1.length - 1, num2[i]));
    }

    console.log(result.join(' '));
    
    process.exit(0);
});